tests/Main.hs

```diff
@@ -33,7 +34,7 @@ testIt name mkTLS getTLS allTLS freeAllTLS = do
   tls <- mkTLS (do putStrLn "  New() called.."
                    newIORef (-1 :: Int))
   mvs <- sequence $ replicate numCap newEmptyMVar
-  forM_ (zip [0..] mvs) $ \(ix,mv) -> forkOn ix $ do
+  forM_ (zip [0..] mvs) $ \(ix,mv) -> forkOS $ do
```

Comment:
That's, uh, kind of a miracle that this was working before. I wonder if it just happened to keep the same OS threads per capability for this one program, and so the TLS managed to work out in the end?